### PR TITLE
Add functionality for teams endpoint

### PR DESCRIPTION
**Pull request by pyasi**
Adding support for the Teams REST endpoint based on the functionality described [here](https://buildkite.com/docs/apis/rest-api/teams#list-teams).

### CODE REVIEW

**Comment by pyasi** [comment]
@lox Would love a review when you get a chance. No rush!

File: buildkite/teams.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
+// Copyright 2014 Mark Wolfe. All rights reserved.
```
Comment (pda) [inline buildkite/teams.go]: This copyright notice doesn't seem accurate.
Perhaps just remove it completely?

Comment (pyasi) [reply buildkite/teams.go]: Yeah I was following what I saw in other files and recent PRs, but agree we should remove it. Thanks!

**Comment by pda** [comment]
Thanks @pyasi :pray:

**Pull request merged**